server/src/test/java/org/opensearch/index/engine/NRTReplicationEngineTests.java

```diff
@@ -75,6 +75,28 @@ public void testCreateEngine() throws IOException {
         }
     }
 
+    public void testCreateEngineWithException() throws IOException {
+        final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.NO_OPS_PERFORMED);
+        final Store nrtEngineStore = createStore(INDEX_SETTINGS, newDirectory());
+        try {
```

Comment:
Nitpick...can use try-with-resources here to ensure things get cleaned up properly even on failure